### PR TITLE
feat(node): update validator reward ratio on L1Block contract

### PR DESCRIPTION
**Pull request by Pangssu**
The name of the Fee Vault has been updated:
- L1FeeVault -> ProposerRewardVault
- ProposerFeeVault -> ValidatorRewardVault
- BaseFeeVault -> ProtocolVault

And, to distribute the transaction fee to the Protocol Vault and the Validator Reward Vault, the ValidatorRewardRatio value in the L1Block contract is updated.

Related: #19

### CODE REVIEW

File: e2e/system_test.go [hunk 1]
```diff
@@ -1318,31 +1318,31 @@ func TestFees(t *testing.T) {
...
-	require.Equal(t, l2Fee, coinbaseDiff, "l2 fee mismatch")
+	require.Equal(t, l2Fee, validatorRewardVaultDiff, "l2 fee mismatch")
 
 	// Tally BaseFee
 	baseFee := new(big.Int).Mul(header.BaseFee, new(big.Int).SetUint64(receipt.GasUsed))
-	require.Equal(t, baseFee, baseFeeRecipientDiff, "base fee fee mismatch")
+	require.Equal(t, baseFee, protocolVaultDiff, "base fee fee mismatch")
 
-	// Tally L1 Fee
+	// Tally V Fee
```
Comment (chokobole) [inline e2e/system_test.go:1368]: What's V Fee?

File: specs/exec-engine.md [hunk 2]
```diff
@@ -8,9 +8,9 @@
   - [Deposited transaction boundaries](#deposited-transaction-boundaries)
 - [Fees](#fees)
   - [Fee Vaults](#fee-vaults)
-  - [Priority fees (Proposer Fee Vault)](#priority-fees-proposer-fee-vault)
-  - [Base fees (Base Fee Vault)](#base-fees-base-fee-vault)
-  - [L1-Cost fees (L1 Fee Vault)](#l1-cost-fees-l1-fee-vault)
+  - [Priority fees (Validator Reward Vault)](#priority-fees-validator-reward-vault)
+  - [Base fees (Protocol Vault)](#base-fees-protocol-vault)
+  - [L1-Cost fees (Proposer Reward Vault)](#l1-cost-fees-proposer-reward-vault)
```
Comment (chokobole) [inline specs/exec-engine.md:13]:
```suggestion
- - [Priority fees (Validator Reward Vault)](#priority-fees-validator-reward-vault)
- - [Base fees (Protocol Vault)](#base-fees-protocol-vault)
- - [L1-Cost fees (Proposer Reward Vault)](#l1-cost-fees-proposer-reward-vault)
+ - [Validator Reward (Validator Reward Vault)](#priority-fees-validator-reward-vault)
+ - [Protocol Fund (Protocol Vault)](#base-fees-protocol-vault)
+ - [Proposer Reward (Proposer Reward Vault)](#l1-cost-fees-proposer-reward-vault)
```

**Comment by chokobole** [comment]
You need to rename `packages/contracts/contracts/test/ProposerFeeVault.t.sol` to `packages/contracts/contracts/test/ValidatorRewardVault.t.sol`

File: packages/contracts/tasks/rekey.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ task('rekey', 'Generates a new set of keys for a test network').setAction(
       'deployer',
       'validator',
       'proxyAdminOwner',
-      'kromaBaseFeeRecipient',
+      'protocolFundRecipient',
       'kromaL1FeeRecipient',
```
Comment (chokobole) [inline packages/contracts/tasks/rekey.ts:14]:
```suggestion
-kromaL1FeeRecipient
+proposerRewardVault
```

File: packages/contracts/tasks/rekey.ts [hunk 4]
```diff
@@ -10,7 +10,7 @@ task('rekey', 'Generates a new set of keys for a test network').setAction(
       'deployer',
       'validator',
       'proxyAdminOwner',
-      'kromaBaseFeeRecipient',
+      'protocolFundRecipient',
```
Comment (chokobole) [inline packages/contracts/tasks/rekey.ts:13]:
```suggestion
-kromaBaseFeeRecipient
+protocolVault
```

**Comment by chokobole** [comment]
I took a look at `e2e/system_test.go` and i found that some comments and semantics are mismatched.

**Comment by chokobole** [comment]
I think that it would be more natural to have the `coinbase` and `suggestedFeeRecipient` point to the `ProposerRewardVault` rather than the `ValidatorRewardVault`.

File: packages/contracts/tasks/rekey.ts (hunk 3 above)
Comment (Pangssu) [reply packages/contracts/tasks/rekey.ts:14]: this is not a vault, withdrawal address.

File: packages/contracts/tasks/rekey.ts (hunk 4 above)
Comment (Pangssu) [reply packages/contracts/tasks/rekey.ts:13]: ditto. not a vault but a withdrawal address.

File: e2e/system_test.go (hunk 1 above)
Comment (Pangssu) [reply e2e/system_test.go:1368]: It's a typo. should be "proposer reward"

**Comment by Pangssu** [comment]
> I think that it would be more natural to have the `coinbase` and `suggestedFeeRecipient` point to the `ProposerRewardVault` rather than the `ValidatorRewardVault`.

In `node/rollup/derive/attributes.go:115`, suggestFeeRecipient is already pointed as ValidatorRewardVault. Can you tell me where it is pointed as ProposerRewardVault?

**Comment by Pangssu** [comment]
> I took a look at `e2e/system_test.go` and i found that some comments and semantics are mismatched.

I have found and fixed. Please let me know if there are any other lines that need to be fixed.

File: specs/exec-engine.md [hunk 5]
```diff
@@ -64,29 +63,28 @@ fee payments are not registered as internal EVM calls, and thus distinguished be
...
-| L1 Fee Vault        | [`L1FeeVault`](./predeploys.md#L1FeeVault)               |
+| Vault Name             | Predeploy                                                      |
+|------------------------|----------------------------------------------------------------|
+| Validator Reward Vault | [`ValidatorRewardVault`](./predeploys.md#ValidatorRewardVault) |
+| Protocol Vault         | [`ProtocolVault`](./predeploys.md#ProtocolVault)               |
+| Proposer Reward Vault  | [`ProposerRewardVault`](./predeploys.md#ProposerRewardVault)   |
 
-### Priority fees (Proposer Fee Vault)
+### Transaction Fees
+Transaction fees in Kroma are different from [eip-1559] specification.
```
Comment (chokobole) [inline specs/exec-engine.md:73]: Could you please put the blank after ###?

File: e2e/system_test.go [hunk 6]
```diff
@@ -9,6 +9,7 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ethereum/go-ethereum"
```
Comment (chokobole) [inline e2e/system_test.go]: Do we need this?

File: e2e/system_test.go [hunk 7]
```diff
@@ -1310,12 +1310,12 @@ func TestFees(t *testing.T) {
 
 	ctx, cancel = context.WithTimeout(context.Background(), 1*time.Second)
 	defer cancel()
-	validatorRewardVaultStartBalance, err := l2Prop.BalanceAt(ctx, header.Coinbase, safeAddBig(header.Number, big.NewInt(-1)))
+	validatorRewardVaultStartBalance, err := l2Prop.BalanceAt(ctx, predeploys.ValidatorRewardVaultAddr, safeAddBig(header.Number, big.NewInt(-1)))
```
Comment (chokobole) [inline e2e/system_test.go:1313]: Does the test remain same semantics..?

File: components/node/rollup/derive/validator_reward.go [hunk 8]
```diff
@@ -0,0 +1,10 @@
+package derive
+
+const TempValidatorRewardRatio = uint64(2000) // 20%
+
+func CalcValidatorRewardRatio() uint64 {
+	// NOTE(pangssu): This is where the ratio for distributing transaction fees as validator rewards is determined.
+	//				  This section requires additional logic to calculate the ratio
+	//		 		  and is currently set to a temporary constant value of 20%.
```
[the remark this answers is not in the document]
Comment (Pangssu) [reply components/node/rollup/derive/validator_reward.go:8]: i got it

File: specs/derivation.md [hunk 9]
```diff
@@ -923,7 +923,7 @@ follows:
 
 - `timestamp` is set to the batch's timestamp.
 - `random` is set to the `prev_randao` L1 block attribute.
-- `suggestedFeeRecipient` is set to the Proposer Fee Vault address. See [Fee Vaults] specification.
+- `suggestedFeeRecipient` is set to the Validator Reward Vault address. See [Fee Vaults] specification.
```
Comment (chokobole) [inline specs/derivation.md:926]:
```suggestion
`suggestedFeeRecipient` is set to the zero address. See [Fee Vaults] specification.
```

File: specs/exec-engine.md [hunk 10]
```diff
@@ -64,29 +63,29 @@ fee payments are not registered as internal EVM calls, and thus distinguished be
...
 
-### Priority fees (Proposer Fee Vault)
+### Transaction Fees
 
-Priority fees follow the [eip-1559] specification, and are collected by the fee-recipient of the L2 block.
-The block fee-recipient (a.k.a. coinbase address) is set to the Proposer Fee Vault address.
+Transaction fees in Kroma are different from [eip-1559] specification.
+The Base Fee is not burned, and there is no distinction between Base Fee and Priority Fee.
+The transaction fee is distributed to two vaults, Validator Reward Vault and Protocol Fee.
+- Validator Reward Vault: `(baseFee + priorityFee) * ValidatorRewardRatio)`
```
Comment (chokobole) [inline specs/exec-engine.md:77]: Please put blank link here!

See `MD032/blanks-around-lists: Lists should be surrounded by blank lines`

**Comment by chokobole** [comment]
We can no longer enjoy the benefits of [EIP-3651](https://eips.ethereum.org/EIPS/eip-3651), as `coinbase` now points to the zero address. As a result, `coinbase.transfer` now means an ETH burn.

However, if we consider PBS (Proposer Builder Separation) in the future, `coinbase` should eventually be set to that of the builder, and `suggestedFeeRecipient` should be the `proposerFeeVault`.

**Review by chokobole — APPROVED** [review]
LGTM

**Comment by Pangssu** [comment]
In the *fuzz test* for `L1Block` contracts, there was an issue where the test would throw an error if `validatorRewardRatio` was greater than `10000`.

**Pull request merged**